app/controllers/media_objects_controller.rb

```diff
@@ -587,6 +581,32 @@ def set_active_file
     return @currentStream
   end
 
+  def update_publish(media_object)
+    errors = []
+    success_count = 0
+    if media_object.title.nil? || media_object.date_issued.nil?
+      errors += ["#{media_object.id}, missing required fields"]
+    else
```

Comment:
I don't think this is necessary. You should already have the error messages in an `ActiveFedora::RecordInvalid` exception when calling `media_object.published!`. We can probably leave the previous code intact and add some error handling to `update_status`.

Reply:
Ohh... I'll check it out :+1: Thanks!